SmartDeviceLink/SDLSoftButtonManager.m

```diff
@@ -293,7 +293,7 @@ - (void)sdl_uploadOtherStateImages {
     for (SDLSoftButtonObject *object in self.softButtonObjects) {
         for (SDLSoftButtonState *state in object.states) {
             if ([state.name isEqualToString:object.currentState.name]) { continue; }
-            if (state.artwork != nil && ![self.fileManager hasUploadedFile:state.artwork]) {
+            if (![self sdl_artworkNeedsUpload:object.currentState.artwork]) {
                 [otherStatesToBeUploaded addObject:state.artwork];
```

Comment:
```suggestion
              [otherStatesToBeUploaded addObject:object.currentState.artwork];
```

Reply:
Otherwise the app crashes when inserting a `nil` `state.artwork` into the array.

Reply:
I think that's incorrect logic that will cause other state artwork to never get uploaded.

Reply:
All I know is that my app kept crashing at that point when I uploaded a soft button with a static icon image